File: Fragment_Communication-II-03Aug21-master/app/src/main/java/com/example/fragmentcommunication_iii03aug21/PreviewActivity.java
package com.example.fragmentcommunication_iii03aug21;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class PreviewActivity extends AppCompatActivity {
    private TextView mTvName, mTvGrade, mTvPercentage, mTvAge;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_preview);
        initViews();
    }

    private void initViews() {
        mTvName = findViewById(R.id.tvStudentName);
        mTvGrade = findViewById(R.id.tvStudentGrade);
        mTvPercentage = findViewById(R.id.tvStudentPercentage);
        mTvAge = findViewById(R.id.tvStudentAge);

        if(getIntent() != null) {
            Model model = (Model) getIntent().getSerializableExtra("model");
            mTvName.setText(model.getName());
            mTvAge.setText(model.getAge()+"");
            mTvGrade.setText(model.getGrade());
            mTvPercentage.setText(model.getPercentage());
        }
    }
}

File: _Kotlin Projects/Navigation_Graph_YOU_20Aug21/app/src/main/java/com/example/navigation_graph_you_20aug21/PreviewActivity.kt
package com.example.navigation_graph_you_20aug21

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_preview.*

class PreviewActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_preview)

        getDataFromIntent()
    }

    private fun getDataFromIntent() {
        intent?.run {
            tvTitle.text = getStringExtra("title")
            tvDescription.text = getStringExtra("description")
            tvStartDate.text = getStringExtra("startDate")
            tvEndDate.text = getStringExtra("endDate")
            tvStartTime.text = getStringExtra("startTime")
            tvEndTime.text = getStringExtra("endTime")
            tvCurrency.text = getStringExtra("currency")
            tvPrice.text = getStringExtra("price")
        }

    }
}